File: Source/Run/SPA/src/app/models/refresh-prices-result.ts
export interface RefreshPriceItemResult {
  investmentId: string;
  name: string;
  ticker?: string;
  status: string;
  message?: string;
  previousAmount?: number;
  newAmount?: number;
  price?: number;
  quoteCurrency?: string;
}

export interface RefreshPricesResult {
  updated: number;
  skipped: number;
  failed: number;
  items: RefreshPriceItemResult[];
}

File: Source/_Core/Contracts/Api/Responses/RefreshPricesResult.cs
namespace WealthLedger.Contracts.Api.Responses;

public class RefreshPricesResult
{
    public int Updated { get; set; }
    public int Skipped { get; set; }
    public int Failed { get; set; }
    public List<RefreshPriceItemResult> Items { get; set; } = [];
}

public class RefreshPriceItemResult
{
    public Guid InvestmentId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Ticker { get; set; }
    public string Status { get; set; } = string.Empty;
    public string? Message { get; set; }
    public decimal? PreviousAmount { get; set; }
    public decimal? NewAmount { get; set; }
    public decimal? Price { get; set; }
    public string? QuoteCurrency { get; set; }
}
